gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/manager/dictionary/impl/I18nDictionaryManagerImpl.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.am.gateway.handler.manager.dictionary.impl;
+
+import io.gravitee.am.common.event.EventManager;
+import io.gravitee.am.common.event.I18nDictionaryEvent;
+import io.gravitee.am.gateway.handler.manager.dictionary.I18nDictionaryManager;
+import io.gravitee.am.gateway.handler.vertx.view.thymeleaf.GraviteeMessageResolver;
+import io.gravitee.am.model.Domain;
+import io.gravitee.am.model.I18nDictionary;
+import io.gravitee.am.model.ReferenceType;
+import io.gravitee.am.model.common.event.Payload;
+import io.gravitee.am.repository.management.api.I18nDictionaryRepository;
+import io.gravitee.common.event.Event;
+import io.gravitee.common.event.EventListener;
+import io.gravitee.common.service.AbstractService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * @author Titouan COMPIEGNE (titouan.compiegne at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class I18nDictionaryManagerImpl extends AbstractService implements I18nDictionaryManager, InitializingBean, EventListener<I18nDictionaryEvent, Payload> {
+
+    private static final Logger logger = LoggerFactory.getLogger(I18nDictionaryManagerImpl.class);
+    private ConcurrentMap<String, I18nDictionary> i18nDictionaries = new ConcurrentHashMap<>();
+
+    @Autowired
+    private I18nDictionaryRepository i18nDictionaryRepository;
+
+    @Autowired
+    private Domain domain;
+
+    @Autowired
+    private EventManager eventManager;
+
+    @Autowired
+    private GraviteeMessageResolver graviteeMessageResolver;
+
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        logger.info("Initializing i18n dictionaries for domain {}", domain.getName());
+        i18nDictionaryRepository.findAll(ReferenceType.DOMAIN, domain.getId())
+                .subscribe(
+                        i18nDictionary -> System.out.println(i18nDictionary),
```

Comment:
Can be replaced with method reference. The error is being logged anyway so I don't think we need `System.out` anyway.

Reply:
FP